hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaSoftwareManagement.java

```diff
@@ -197,14 +192,18 @@ public SoftwareModule createSoftwareModule(final SoftwareModuleCreate c) {
     }
 
     @Override
-    public SoftwareModule findSoftwareModuleById(final Long id) {
-        return softwareModuleRepository.findOne(id);
+    public Optional<SoftwareModule> findSoftwareModuleById(final Long id) {
+        return Optional.ofNullable(softwareModuleRepository.findOne(id));
     }
 
     @Override
-    public SoftwareModule findSoftwareModuleByNameAndVersion(final String name, final String version,
+    public Optional<SoftwareModule> findSoftwareModuleByNameAndVersion(final String name, final String version,
             final Long typeId) {
 
+        if (!softwareModuleTypeRepository.exists(typeId)) {
```

Comment:
This code-block is used 7 times in this class, maybe refactor this in a separate method.
```
if (!softwareModuleTypeRepository.exists(typeId)) {
   throw new EntityNotFoundException(SoftwareModuleType.class, typeId);
}
```